walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,101 @@
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public Collection<TimeRange> query(Collection<Event> events, 
+                                     MeetingRequest request) {
+    // Get unavailable time for all required attendees in meeting request
+    ArrayList<TimeRange> unavailable = new ArrayList<>();
+    for (String attendee : request.getAttendees()) {
+      ArrayList<TimeRange> meetingTimes = getMeetingTimes(events, attendee);
+      unavailable = merge(unavailable, meetingTimes);
+    }
+
+    // Get time ranges that are longer than the requested meeting duration
+    Collection<TimeRange> result = getSortedAvailability(unavailable);
+    result.removeIf( 
+      timeRange -> timeRange.duration() < request.getDuration());
+
+    return result;
+  }
+
+  /**
+   * Adds the unavailable time ranges of a new attendee to the cumulative
+   * unavailable time ranges.
+   * Resolves conflicts by merging overlapping ranges into new ranges
+   */
+  private ArrayList<TimeRange> merge(ArrayList<TimeRange> result, 
+                                     ArrayList<TimeRange> other) {
+    if (result.isEmpty()) {
+      return other;
+    }
+    if (other.isEmpty()) {
+      return result;
+    }
+
+    result.addAll(other);
+    Collections.sort(result, TimeRange.ORDER_BY_START);
+
+    // Merge overlapping ranges
+    for (int i = 0; i < result.size()-1; i++) {
+      TimeRange current = result.get(i);
+      TimeRange next = result.get(i+1);
+
+      if (current.overlaps(next)) {
+        if (current.contains(next)) {
+          // Case 1:  |_______|
+          //            |__|
+          result.remove(next);
+        } else {
+          // Case 2:  |______|
+          //               |______|
+          result.set(i, TimeRange.fromStartEnd(current.start(), 
+                                               next.end(), false));
+          result.remove(next);
+          i--;
+        }
+      }
+    }
+    
+    return result;
+  }
+
+  /** Gets an attendee's unavailable time ranges */
+  private ArrayList<TimeRange> getMeetingTimes(Collection<Event> events, 
+                                               String attendee) {
+    ArrayList<TimeRange> meetingTimes = new ArrayList<>();
+    for (Event e : events) {
+      if (e.getAttendees().contains(attendee)) {
+        meetingTimes.add(e.getWhen());
+      }
+    }
+
+    return meetingTimes;
+  }
+
+  /**
+   * @param sortedUnavailable Attendees' unavailable time ranges sorted 
+   *                          by start time in ascending order
+   * @return Attendees' available time ranges sorted by start time in 
+   *         ascending order
+   */
+  private Collection<TimeRange> getSortedAvailability(Collection<TimeRange> 
```

Comment:
you should not break a line at variable type & variable name.

optional:
And the function name sounds like it returns sorted result rather than requiring sorted input.
Just keep the original function name ```getAvailability``` and change the parameter name to ```sortedUnavailable``` is enough.